cmake/JSSTests.cmake

```diff
@@ -202,6 +209,30 @@ macro(jss_tests)
     )
 endmacro()
 
+macro(jss_tests_compile)
+    jss_tests_compile_c("${PROJECT_SOURCE_DIR}/org/mozilla/jss/tests/TestBufferPRFD.c" "${BIN_OUTPUT_DIR}/TestBufferPRFD" "TestBufferPRFD")
+endmacro()
+
+macro(jss_tests_compile_c C_FILE C_OUTPUT C_TARGET)
+    # Generate the target executable from C_FILE
+
+    add_custom_command(
+        OUTPUT "${C_OUTPUT}"
+        COMMAND ${CMAKE_C_COMPILER} ${JSS_C_FLAGS} -o ${C_OUTPUT} ${C_FILE} ${JSS_LD_FLAGS}
```

Comment:
Ubuntu was using an older GCC compiler, which strictly requires linker flags to be at the end.